vimiv/api/settings.py

```diff
@@ -488,3 +488,14 @@ class title:  # pylint: disable=invalid-name
         desc="Default window title if no mode specific options exist",
     )
     StrSetting("title.image", "vimiv - {basename}", desc="Window title in image mode")
+
+
+class metadata:  # pylint: disable=invalid-name
+    """Namespace for metadata related settings."""
+
+    # Store the keys as a comma seperated string
+    StrSetting(
```

Comment:
It would be great to asign the setting to a variable, e.g.: `keylist = StrSetting( ...`. Like this we can refer to it later via the python object, not only as a string.

Reply:
Surely, I have noticed that you sometimes assign the setting to a variable and sometimes you dont. May I ask you want the advantage or point of doing in one or the other way?

Reply:
The general rule is to use the variable where possible as we can then make use of actual python objects directly. This also gives us access to the `updated` signal which allows objects to react when a setting changed, e.g. the slideshow in `imutils.slideshow` has a timer that updates its interval upon `api.settings.slideshow.delay.changed`.

Of course there are places where we cannot use the type immediately, for example when running the `:set` command on the command-line, we first retrieve the object by its name as only the name can be passed as a string by the user.

If there are only cases where the setting is used by name and it cannot be used by its type, the setting is not assigned to al variable. This is currently only the case for status-related settings, for example `statusbar.left` and `statusbar.left_image`. Here we can define arbitrary combinations of `position_mode`, but not all have defaults. We also only retrieve the value via this string pattern as we can then iterate over the positions and make use of the current mode name. This happens for example in `gui.statusbar` in the `_get_text(position, mode)` method.

tl;dr assign where possible, use string where the code is too dynamic to access the variable directly

Hope this makes it a bit more clear instead of more confusing :smile: